exercise3/node.py

```diff
@@ -25,53 +24,29 @@ class Node:
     def __init__(self, owner_public_key: PublicKey, initial_transaction: Transaction):
         """
         TODO: Przypisz wartości polom owner oraz blockchain przy pomocy podanych argumentów.
-        Wykorzystaj `initial_transaction` do stworzenia blockchain (hash poprzedniego bloku i nonce powinny być zerem).
-
+        Wykorzystaj `initial_transaction` do stworzenia blockchain.
         """
         self.owner = owner_public_key
-        self.blockchain = Blockchain(
-            [Block(b"\x00", int(time()), 0, [initial_transaction])]
-        )
+        self.blockchain = Blockchain(initial_transaction)
 
-    def add_transaction(self, transaction: Transaction):
-        """
-        TODO: Dodaj podaną transakcję do bloku.
-        Sprawdź, czy transakcja jest poprawna (użyj metody `validate_transaction`), jeśli nie jest, rzuć wyjątek.
-        Stwórz transakcję generującą nowego coin'a, aby wynagrodzić właściciela node'a.
-        Stwórz nowy blok zawierający obie transakcje.
-        Znajdź nonce, który spełni kryteria sieci (użyj metody `find_nonce`).
-        Dodaj blok na koniec łańcucha.
+    def validate_transaction(self, transaction: Transaction) -> bool:
         """
-        if not self.validate_transaction(transaction):
-            raise Exception("Transaction can't be added. Verification failed.")
+        TODO: Sprawdź poprawność transakcji.
+        Transakcja jest poprawna, jeśli:
+        - ma podpis,
+        - podpis jest poprawny
```

Comment:
```suggestion
        - podpis jest poprawny (...)
```
co to znaczy że jest poprawny, można by rozwinąć kto miał to podpisać